src/core/search/parser.y

```diff
@@ -98,15 +101,15 @@ final_query:
 
 knn_query:
   LBRACKET KNN UINT32 FIELD TERM opt_knn_alias opt_ef_runtime RBRACKET
-    { $$ = AstKnnNode($3, $4, BytesToFtVector($5), $6, $7); }
+    { $$ = AstKnnNode(toUint32($3), $4, BytesToFtVector($5), $6, $7); }
```

Comment:
I think it would be better to create expression UINT32 and DOUBLE as not to write each time `toUint32` / `toDouble`

Reply:
What do you mean

Reply:
to add something like `%nterm <uint32_t> UINT_VALUE' that returns `toUint32($1)`  and use this nterm in other expression. To not to write each time toUint32

same for double

Reply:
it was implemented as %nterm <uint32_t> UINT_VALUE the problem that we don't know the final type that we need in lexer.

Reply:
```
uint_value: UINT32 { $$ = toUint32($1); }

double_value: DOUBLE { $$ = toDouble($1); }
```
Upd.: after discussion, it was decided that these rules are redundant